stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/search/SearchResultListAdapter.kt

```diff
@@ -0,0 +1,64 @@
+package io.getstream.chat.android.ui.search
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.ui.databinding.StreamItemMentionListBinding
+import io.getstream.chat.android.ui.messagepreview.MessagePreviewView
+import io.getstream.chat.android.ui.search.SearchResultListAdapter.MessagePreviewViewHolder
+import io.getstream.chat.android.ui.search.SearchResultListView.SearchResultSelectedListener
+
+public class SearchResultListAdapter : ListAdapter<Message, MessagePreviewViewHolder>(MessageDiffCallback) {
+
+    private var searchResultSelectedListener: SearchResultSelectedListener? = null
```

Comment:
What do you think about making it non-nullable and set an empty listener by default?

Reply:
It is! It's `null` by default anyway, and this way you can unset it by passing in `null`, if you want to.

Reply:
It's easier to clear the listener this way, otherwise you'd have to pass in your own empty implementation to stop listening to events. Same behaviour as `View#setOnClickListener`, which also allows you to set to `null` to clear it.